retrievalmarket/impl/provider.go

```diff
@@ -109,6 +114,7 @@ func NewProvider(minerAddress address.Address,
 	dataTransfer datatransfer.Manager,
 	ds datastore.Batching,
 	retrievalPricingFunc RetrievalPricingFunc,
+	piecesWithBlock GetPiecesContainingBlock,
```

Comment:
Why don't we make this a method on the DAGStoreWrapper rather than adding on more abstraction here ?

Reply:
Makes sense 👍